370-auth-ui/oidc.md

```diff
@@ -0,0 +1,351 @@
+
+## Integrate Skygear Auth OIDC (Skygear App as OIDC Auth provider + External Application)
+
+### Example Scenarios
+
+- Independent backend connect with Skygear App as auth provider
+- Developer has microservices in Skygear App A which connect with Skygear App B as auth provider
+- Independent gateway connect with Skygear App as auth provider
+
+### Generic OIDC RP
+
+[![](https://mermaid.ink/img/eyJjb2RlIjoic2VxdWVuY2VEaWFncmFtXG4gIHBhcnRpY2lwYW50IENsaWVudEFwcFxuICBwYXJ0aWNpcGFudCBBcHBCYWNrZW5kXG4gIHBhcnRpY2lwYW50IFNreWdlYXJBdXRoXG4gIENsaWVudEFwcC0-PkFwcEJhY2tlbmQ6IFVzZXIgY2xpY2sgbG9naW5cbiAgQXBwQmFja2VuZC0-PlNreWdlYXJBdXRoOiBBdXRob3JpemF0aW9uIGNvZGUgcmVxdWVzdFxuICBTa3lnZWFyQXV0aC0-PkNsaWVudEFwcDogUmVkaXJlY3QgdG8gYXV0aG9yaXphdGlvbiBwcm9tb3RlXG4gIENsaWVudEFwcC0-PlNreWdlYXJBdXRoOiBBdXRob3JpemF0aW9uIGFuZCBjb25zZW50XG4gIFNreWdlYXJBdXRoLT4-QXBwQmFja2VuZDogQXV0aG9yaXphdGlvbiBjb2RlXG4gIEFwcEJhY2tlbmQtPj5Ta3lnZWFyQXV0aDogQXV0aG9yaXphdGlvbiBjb2RlICsgY2xpZW50IGlkICsgY2xpZW50IHNlY3JldFxuICBTa3lnZWFyQXV0aC0-PlNreWdlYXJBdXRoOiBWYWxpZGF0ZSBhdXRob3JpemF0aW9uIGNvZGUgKyBjbGllbnQgaWQgKyBjbGllbnQgc2VjcmV0XG4gIFNreWdlYXJBdXRoLT4-QXBwQmFja2VuZDogVG9rZW4gcmVzcG9uc2UgKElEIHRva2VuICsgYWNjZXNzIHRva2VuICsgcmVmcmVzaCB0b2tlbilcbiAgQXBwQmFja2VuZC0-PlNreWdlYXJBdXRoOiBSZXF1ZXN0IHVzZXIgZGF0YSB3aXRoIGFjY2VzcyB0b2tlblxuICBTa3lnZWFyQXV0aC0-PkFwcEJhY2tlbmQ6IFJlc3BvbnNlIHVzZXIgZGF0YVxuICBBcHBCYWNrZW5kLT4-QXBwQmFja2VuZDogQ3JlYXRlIEFwcEJhY2tlbmQgbWFuYWdlZCBzZXNzaW9uXG4gIEFwcEJhY2tlbmQtPj5DbGllbnRBcHA6IFJldHVybiBBcHBCYWNrZW5kIG1hbmFnZWQgc2Vzc2lvbiIsIm1lcm1haWQiOnsidGhlbWUiOiJkZWZhdWx0Iiwic2VxdWVuY2UiOnsic2hvd1NlcXVlbmNlTnVtYmVycyI6dHJ1ZX19LCJ1cGRhdGVFZGl0b3IiOmZhbHNlfQ)](https://mermaid-js.github.io/mermaid-live-editor/#/edit/eyJjb2RlIjoic2VxdWVuY2VEaWFncmFtXG4gIHBhcnRpY2lwYW50IENsaWVudEFwcFxuICBwYXJ0aWNpcGFudCBBcHBCYWNrZW5kXG4gIHBhcnRpY2lwYW50IFNreWdlYXJBdXRoXG4gIENsaWVudEFwcC0-PkFwcEJhY2tlbmQ6IFVzZXIgY2xpY2sgbG9naW5cbiAgQXBwQmFja2VuZC0-PlNreWdlYXJBdXRoOiBBdXRob3JpemF0aW9uIGNvZGUgcmVxdWVzdFxuICBTa3lnZWFyQXV0aC0-PkNsaWVudEFwcDogUmVkaXJlY3QgdG8gYXV0aG9yaXphdGlvbiBwcm9tb3RlXG4gIENsaWVudEFwcC0-PlNreWdlYXJBdXRoOiBBdXRob3JpemF0aW9uIGFuZCBjb25zZW50XG4gIFNreWdlYXJBdXRoLT4-QXBwQmFja2VuZDogQXV0aG9yaXphdGlvbiBjb2RlXG4gIEFwcEJhY2tlbmQtPj5Ta3lnZWFyQXV0aDogQXV0aG9yaXphdGlvbiBjb2RlICsgY2xpZW50IGlkICsgY2xpZW50IHNlY3JldFxuICBTa3lnZWFyQXV0aC0-PlNreWdlYXJBdXRoOiBWYWxpZGF0ZSBhdXRob3JpemF0aW9uIGNvZGUgKyBjbGllbnQgaWQgKyBjbGllbnQgc2VjcmV0XG4gIFNreWdlYXJBdXRoLT4-QXBwQmFja2VuZDogVG9rZW4gcmVzcG9uc2UgKElEIHRva2VuICsgYWNjZXNzIHRva2VuICsgcmVmcmVzaCB0b2tlbilcbiAgQXBwQmFja2VuZC0-PlNreWdlYXJBdXRoOiBSZXF1ZXN0IHVzZXIgZGF0YSB3aXRoIGFjY2VzcyB0b2tlblxuICBTa3lnZWFyQXV0aC0-PkFwcEJhY2tlbmQ6IFJlc3BvbnNlIHVzZXIgZGF0YVxuICBBcHBCYWNrZW5kLT4-QXBwQmFja2VuZDogQ3JlYXRlIEFwcEJhY2tlbmQgbWFuYWdlZCBzZXNzaW9uXG4gIEFwcEJhY2tlbmQtPj5DbGllbnRBcHA6IFJldHVybiBBcHBCYWNrZW5kIG1hbmFnZWQgc2Vzc2lvbiIsIm1lcm1haWQiOnsidGhlbWUiOiJkZWZhdWx0Iiwic2VxdWVuY2UiOnsic2hvd1NlcXVlbmNlTnVtYmVycyI6dHJ1ZX19LCJ1cGRhdGVFZGl0b3IiOmZhbHNlfQ)
```

Comment:
Typo. "Redirect to authorization promote" -> "Redirect to authorization endpoint". Apply to other flow chart as well.